src/ray/common/task/task_spec.h

```diff
@@ -36,7 +36,30 @@ extern "C" {
 namespace ray {
 inline bool operator==(const ray::rpc::SchedulingStrategy &lhs,
                        const ray::rpc::SchedulingStrategy &rhs) {
-  return google::protobuf::util::MessageDifferencer::Equals(lhs, rhs);
+  if (lhs.scheduling_strategy_case() != rhs.scheduling_strategy_case()) {
+    return false;
+  }
+
+  switch (lhs.scheduling_strategy_case()) {
+  case ray::rpc::SchedulingStrategy::kNodeAffinitySchedulingStrategy: {
```

Comment:
since we are customizing the equal check, should we add unit tests? :)

Reply:
We did have tests covering these already: 
https://github.com/ray-project/ray/blob/50e278f58bc10be699bb154875231b7a9b79e429/src/ray/common/test/task_spec_test.cc#L20-L101

(it's on the `SchedlingClassDescriptor` equality check - but it's testing all the sub-fields equality checks).